Base.rte/AI/HumanBehaviors.lua

```diff
@@ -1356,15 +1365,19 @@ function HumanBehaviors.GoToWpt(AI, Owner, Abort)
 								
 								if digState == AHuman.NOTDIGGING and AI.deviceState ~= AHuman.DIGGING then
 									-- if our path isn't blocked enough to dig, but the headroom is too little, start crawling to get through
-									local Heading = SceneMan:ShortestDistance(Owner.Pos, Waypoint.Pos, false):SetMagnitude(Owner.Height*0.5)
+									local heading = SceneMan:ShortestDistance(Owner.Pos, Waypoint.Pos, false):SetMagnitude(Owner.Height*0.5)
 									
-									-- don't crawl if it's too steep, climb then instead
-									if math.abs(Heading.X) > math.abs(Heading.Y) and Owner.Head and Owner.Head:IsAttached() then
-										local TopHeadPos = Owner.Head.Pos - Vector(0, Owner.Head.Radius*0.7)
+									local angleRadians = math.abs(math.atan2(-(heading.X * heading.Y), heading.X * heading.X));
+									local angleDegrees = angleRadians * (180 / math.pi);
```

Comment:
We have `Vector::AbsDegAngle` for this. Assuming this is just getting the vector's angle, I'd rather use the more readable method than the in-lua calculation here.

Reply:
Sorry, I've forgotten too much maffs over the years (and also it's late and I'm tired) so this one has taken me a bit to sort out, but I'm hoping I'm not too far off base.

First off, I wanna be sure I'm understanding, you're getting the heading (which I assume is the vector describing the direction the actor is currently moving in) and considering its angle but between -90 and 270 instead of 0 and 360, right?
Given that, you can abs it and then any angle < 30 is relatively flat, if it's facing right. That makes sense to me, but how does facing left work?

Also, your way is probably cheaper, but I did recently add angle helper methods in RTETools (and hopefully I didn't make any mistakes that can cause problems in edge cases haha) for things like changing angles from 0 - 360 to -180 - 180, and for checking if an angle is within a range and stuff. I guess these would be reasonable ways to do this with built-in functions, but as I said, yours is probably cheaper.

Reply:
Sorry, let me explain;

This Lua code gets the angle of the heading vector relative to flat (i.e, straight along the X axis). The direction along the X axis doesn't matter. As such, this gives a potential value of -90 to 90, which is then surrounded in math,.abs() to get a value between 0 and 90. 0 is pointing straight left/right, and 90 is pointing straight up/down.

VectorAbsDegAngle(), according to it's documentation, returns a value representing the vectors absolute angle:
`		/// <summary>
		/// Get this Vector's absolute angle in degrees. e.g: when x = 1, y = 0, the value returned here will be 0. x = 0, y = 1 yields -90 here.
		/// </summary>
		/// <returns>The absolute angle in degrees, in the interval [-90, 270).</returns>`

And there's no code involved or mentioned here that gives or needs an angle between 0 and 360.

Reply:
Holy shit, I don't think I ever noticed that AbsRadAngle/AbsDegAngle is range [-90 to 270]. I guess it's been fine, but that seems like an odd choice!

Given the [-90, 270) range is from AbsDegAngle, this makes sense now.
As you say, a Vector angle comparison method (or maybe 2 so you can compare with a rad angle directly) would be a good thing to have in general, but that can be a future bit of work. For now, would you mind chucking in a brief explanatory comment, just clarifying the range that this bit of math spits out, so someone else looking at it can figure out what's going on?